### PR TITLE
feat(autocomplete): implement Autocomplete component

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Autocomplete**
Github issue: #489 

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [x] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: packages/components/src/core/Autocomplete/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      require: true,
+    },
+  },
+  component: Autocomplete,
+  // (masoudmanson) For the purpose of storybook, the button is removed
+  // from the dropdown menu component which may cause some accessibility
+  // violations related to ARIA roles and attributes. However, this
+  // should not be a concern as the component is always used with a button
+  // in real applications. To avoid false positive test failures, the following
+  // accessibility rules have been temporarily disabled in the tests
```
Comment (tihuan) [inline packages/components/src/core/Autocomplete/index.stories.tsx:88]: thanks for the explanation!

**Review by tihuan — APPROVED** [review]
Thanks for the component, Masoud!

The only UI feedback I have is that IIRC Harley wants the input text to be cleared once a selection is made 🙆‍♂️ 

Screenshot 2023-08-04 at 7 21 07 AM

Thanks again!

**Comment by tihuan** [comment]
Fantastic 🎉 Thanks again, Masoud! LGTMM

@hthomas-czi @clarsen-czi  PTAL too thank you!

https://www.chromatic.com/component?appId=616981d9028812003ade73f3&csfId=dropdowns-autocomplete&buildNumber=1445&k=64d536b8de00f92bf72cb131-1200-interactive-true&h=7&b=-1

**Review by hthomas-czi — APPROVED** [review]
Looks good, Masoud!

**Pull request merged**